src/js/components/Hero.js

```diff
@@ -0,0 +1,148 @@
+    }
+  }
+
+  _setReverse () {
+    const { justify } = this.props;
+
+    if (window.innerWidth < PALM_BREAKPOINT) {
+      this.setState({ reverse: false });
+    } else {
+      this.setState({ reverse: (justify === 'start') ? true : false });
+    }
+  }
+
+  render () {
+    const { backgroundImage, backgroundType, backgroundVideoLoop, backgroundVideoMuted, backgroundVideoPoster, backgroundVideoSource, children, className, flush, image, justify, responsiveBackgroundPosition, separator, size } = this.props;
+
+    let classes = classnames(
+      CLASS_ROOT,
+      className,
+      {
+        [`${CLASS_ROOT}--${size}`]: size,
+        [`${CLASS_ROOT}--bg-${responsiveBackgroundPosition}`]: responsiveBackgroundPosition,
+        [`${CLASS_ROOT}--mobile-separator`]: separator
+      }
+    );
+
+    let full = flush ? 'horizontal' : false;
+    let pad = flush ? 'none' : 'large';
+
+    let backgroundMarkup;
+    if (backgroundType === 'image') {
+      backgroundMarkup = <Box containerClassName={CLASS_ROOT + "__background"} appCentered={true} pad={pad} backgroundImage={`url(${backgroundImage})`} full={full} />;
+    } else if (backgroundType === 'video') {
+      backgroundMarkup = (
+        <Video className={CLASS_ROOT + "__background"} autoPlay={true} showControls={false} loop={backgroundVideoLoop} muted={backgroundVideoMuted} poster={backgroundVideoPoster}>
+          <source src={backgroundVideoSource} type="video/mp4"/>
+        </Video>
+      );
+    }
+
+    let imageMarkup = <Box />;
+    if (image) {
+      imageMarkup = <Image src={`url(${image})`} />;
+    }
+
+    let contentMarkup;
+    if (justify === 'center') {
+      contentMarkup = (
+        <Box className={CLASS_ROOT + "__overlay"} justify={justify} align="center" primary={true} full={full} direction="row" >
+          <Box pad={{horizontal: 'large', vertical: 'large', between: 'medium'}}>
+            {children}
+          </Box>
+        </Box>
+      );
+    } else {
+      contentMarkup = (
+        <Box className={CLASS_ROOT + "__overlay"} align="center" primary={true} full={full} direction="row" reverse={this.state.reverse} >
+          <Box className={CLASS_ROOT + "__image"} align="center" justify="center">
+            {imageMarkup}
+          </Box>
+          <Box pad={{horizontal: 'large', vertical: 'large', between: 'medium'}}>
+            {children}
+          </Box>
+        </Box>
+      );
+    }
+
+    return (
+      <Box className={classes} colorIndex={this.state.colorIndex}>
+        {backgroundMarkup}
+        {contentMarkup}
+      </Box>
+    );
+  }
+}
+
+Hero.propTypes = {
+  backgroundImage: PropTypes.string,
+  backgroundType: PropTypes.oneOf(['image', 'video']),
+  backgroundVideoLoop: PropTypes.bool,
+  backgroundVideoMuted: PropTypes.bool,
+  backgroundVideoPoster: PropTypes.string,
+  backgroundVideoSource: PropTypes.string,
+  darkTheme: PropTypes.bool,
```

Comment:
Should we allow colorIndex as another type of background in addition to image and video?